core/controllers/platform_feature_test.py

```diff
@@ -78,17 +81,24 @@ def setUp(self):
             ]
         )
 
-        feature_services.ALL_FEATURES_LIST = param_names
+        # Here we use MyPy ignore because the expected type of ALL_FEATURES_LIST
+        # is a list of 'PARAM_NAMES' Enum, but here for testing purposes we are
+        # providing a list of 'ParamNames' enums, which causes MyPy to throw an
```

Comment:
This is now a bit confusing and potentially reads as a contradiction. What's the difference between a ParamName and a PARAM_NAME?

Reply:
Comment updated, PTAL!

Basically, here ALL_FEATURES_LIST is expected to be a list of `platform_feature_list.ParamNames` enums https://github.com/oppia/oppia/blob/cc1376ab305ef647b2c8f7a817c5f1e595ed0661/core/domain/platform_feature_services.py#L42
but here for testing purposes, we are defining our own custom ParamNames enums and assigning the list of its elements to the ALL_FEATURES_LIST:
https://github.com/oppia/oppia/blob/cc1376ab305ef647b2c8f7a817c5f1e595ed0661/core/controllers/platform_feature_test.py#L30

and according to MyPy, `platform_feature_list.ParamNames` enum is different from platform_feature_test's ParamNames enum.

> What's the difference between a ParamName and a PARAM_NAME?

Sorry, PARAM_NAME was added accidently. it should be `platform_feature_list.ParamNames`

Reply:
Ah I see--this is much clearer now. Thanks!